base/workerpool.jl

```diff
@@ -0,0 +1,85 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+type WorkerPool
+    channel::RemoteChannel{Channel{Int}}
+end
+
+
+isbusy(pool::WorkerPool) = !isready(pool.channel)
+
+
+"""
+    WorkerPool(workers)
+
+Create a WorkerPool from a vector of worker ids.
+"""
+function WorkerPool(workers::Vector{Int})
+
+    # Create a shared queue of workers...
+    pool = RemoteChannel(()->Channel{Int}(128)) 
+
+    # Check that workers are not already part of a pool...
+    check = () -> if :_worker_pool in names(Main)
+        error("Worker $(myid()) already in a WorkerPool!")
+    end
+    foreach(fetch, [@spawnat w check() for w in workers])
+
+    # Put each worker into the pool...
+    for w in workers
+        put!(pool, w)
+        @spawnat w global _worker_pool = pool
+    end
+
+    WorkerPool(pool)
+end
+
+
+"""
+    WorkerPool(n)
+
+Launches `n` workers using the in-built LocalManager.
+"""
+WorkerPool(n::Integer) = WorkerPool(addprocs(n))
+
+
+"""
+    WorkerPool()
+
+Equivalent to WorkerPool(CPU_CORES)
+"""
+WorkerPool() = WorkerPool(addprocs())
+
+
+"""
+    remotecall_fetch(f, pool::WorkerPool, args...)
+
+Call `f(args...)` on one of the workers in `pool`.
+"""
+function remotecall_fetch(f, pool::WorkerPool, args...)
+    l = (args...)->try f(args...) finally put!(_worker_pool, myid()) end
+    remotecall_fetch(l, take!(pool.channel), args...)
+end
```

Comment:
The `put!(_worker_pool, myid())` will result in a separate message sent to the caller. This can be optimized by doing it on the caller.

```
function remotecall_fetch(f, pool::WorkerPool, args...)
     pid = take!(pool.channel)
     try
          remotecall_fetch(f, pid, args...)
     finally
          put!(pool.channel, pid) 
     end
 end
```